hamlpy/compiler.py

```diff
@@ -8,16 +8,48 @@
 from hamlpy.parser.nodes import Node, read_node
 
 
-class Compiler:
-    DEFAULT_OPTIONS = {
-        'attr_wrapper': '\'',            # how to render attribute values, e.g. foo='bar'
-        'django_inline_style': False,    # support both #{...} and ={...}
-        'tag_config': 'django',          # Django vs Jinja2 tags
-        'custom_self_closing_tags': {},  # additional self-closing tags
-        'cdata': True,                   # wrap CSS, Javascript etc content in CDATA section
-        'debug_tree': False,
-    }
+class Options(object):
+    HTML4 = 'html4'
+    HTML5 = 'html5'
+    XHTML = 'xhtml'
+
+    def __init__(self, **kwargs):
+        # standard Haml options
+        self.attr_wrapper = '\''  # how to render attribute values, e.g. foo='bar'
+        self.format = self.HTML5  # HTML4, HTML5 or XHTML
+        self.escape_attrs = True  # escape HTML-sensitive characters in attribute values
+        self.cdata = True         # wrap CSS, Javascript etc content in CDATA section
+
+        # implementation specific options
+        self.django_inline_style = False    # support both #{...} and ={...}
+        self.tag_config = 'django'          # Django vs Jinja2 tags
+        self.custom_self_closing_tags = {}  # additional self-closing tags
+        self.debug_tree = False
+
+        for k, v in kwargs.items():
+            setattr(self, k, v)
+
+        if self.django_inline_style:  # pragma: no cover
+            warnings.warn("Support for ={..} style variables is deprecated", DeprecationWarning)
+
+    @property
+    def html4(self):
+        return self.format == self.HTML4
+
+    @property
+    def html5(self):
+        return self.format == self.HTML5
+
+    @property
+    def html(self):
+        return self.html4 or self.html5
```

Comment:
This pattern of using properties lets us follow the Ruby Haml code a bit more closely